CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ReplicatorConfigurationMethods.cs

```diff
@@ -0,0 +1,309 @@
+        }
+
+        public static void SetAuthenticator([NotNull] NameValueCollection args,
+                             [NotNull] IReadOnlyDictionary<string, object> postBody,
+                             [NotNull] HttpListenerResponse response)
+        {
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                With<Authenticator>(postBody, "authenticator", auth => {
+                    repConf.Authenticator = auth;
+                    response.WriteEmptyBody();
+                });
+            });
+        }
+
+        public static void SetChannels([NotNull] NameValueCollection args,
+                     [NotNull] IReadOnlyDictionary<string, object> postBody,
+                     [NotNull] HttpListenerResponse response)
+        {
+            IList<string> channels = (IList<string>)postBody["channels"];
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                repConf.Channels = channels;
+            });
+        }
+
+        public static void SetContinuous([NotNull] NameValueCollection args,
+             [NotNull] IReadOnlyDictionary<string, object> postBody,
+             [NotNull] HttpListenerResponse response)
+        {
+            Boolean continuous = (Boolean)postBody["continuous"];
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                repConf.Continuous = continuous;
+            });
+        }
+
+        public static void SetDocumentIDs([NotNull] NameValueCollection args,
+             [NotNull] IReadOnlyDictionary<string, object> postBody,
+             [NotNull] HttpListenerResponse response)
+        {
+            IList<string> documentIds = (IList<string>)postBody["documentIds"];
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                repConf.DocumentIDs = documentIds;
+            });
+        }
+
+        public static void SetReplicatorType([NotNull] NameValueCollection args,
+                     [NotNull] IReadOnlyDictionary<string, object> postBody,
+                     [NotNull] HttpListenerResponse response)
+        {
+            var replicatorType = postBody["replType"].ToString().ToLower();
+            ReplicatorType replType = new ReplicatorType();
+
+            if (replicatorType == "push")
+            {
+                replType = ReplicatorType.Push;
+            }
+            else if (replicatorType == "pull")
+            {
+                replType = ReplicatorType.Pull;
+            }
+            else
+            {
+                replType = ReplicatorType.PushAndPull;
+            }
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                repConf.ReplicatorType = replType;
+            });
+            response.WriteEmptyBody();
+        }
+
+        // TODO have to fix this method
```

Comment:
I hope this method fix with your latest changes. If yes, do you want to remove this TODO?

Reply:
No, I haven't fixed this method, as I was not able to find the way to convert var to byte[].